packages/block-editor/src/hooks/style.js

```diff
@@ -58,7 +47,10 @@ const VARIABLE_REFERENCE_PREFIX = 'var:';
 const VARIABLE_PATH_SEPARATOR_TOKEN_ATTRIBUTE = '|';
 const VARIABLE_PATH_SEPARATOR_TOKEN_STYLE = '--';
 function compileStyleValue( uncompiledValue ) {
-	if ( startsWith( uncompiledValue, VARIABLE_REFERENCE_PREFIX ) ) {
+	if (
+		typeof uncompiledValue === 'string' &&
+		uncompiledValue.startsWith( VARIABLE_REFERENCE_PREFIX )
```

Comment:
We could use an optional operator here instead, it's used in multiple places already. 
```suggestion
		uncompiledValue?.startsWith?.( VARIABLE_REFERENCE_PREFIX )
```